### ISSUE TITLE
Fails to publish built resources to gh-pages

### ISSUE THREAD

**Issue by oleersoy** [travis-ci/travis-ci#7490]
I did a writeup of the issue on SO [here](http://stackoverflow.com/questions/42869506/publishing-built-files-to-gh-pages-with-travis-ci). 

I followed the https://docs.travis-ci.com/user/deployment/pages/ for the setup.

**Comment by BanzaiMan** [comment]
Do you have a build log URL that shows the problem you are describing here?

**Comment by oleersoy** [comment]
I can lookup how to get one, but I'm not sure it will show anything.  The configuration is working ... sort of ... for example here's the result of a recent commit:
https://github.com/superfly-css/superfly-css-component-button/tree/gh-pages

The `src` directory containing all the source files are being pushed to `gh-pages`.  However the `target` folder that the build creates is not being pushed.  I think it might be because my .gitignore file contains the `target` folder (I'll check in a moment).

This is how I have travis configured ATM:
https://github.com/superfly-css/superfly-css-component-button/blob/master/.travis.yml

**Comment by oleersoy** [comment]
OK - Yes - it works now - the problem was that the `target` folder was listed in .gitignore.  However if it's not in `.gitignore` then someone might accidentally commit build files ... ideally travis would have a configuration option for the built folder(s) that should be committed to `gh-pages`.

**Comment by BanzaiMan** [comment]
Could you try modifying `.gitignore` in the `before_deploy` phase?

For example:

```yaml
before_deploy:
  - sed -i '/target/d' .gitignore
```

**Comment by oleersoy** [comment]
That sounds like it would do the trick.  I just added `target` back to `.gitignore` and update the travis configuration with the snippet.  I don't an update to `gh-pages` yet, but maybe it takes a while?

**Comment by oleersoy** [comment]
YES!!  It worked.  I see the update now.  Brilliant - Thanks!

**Comment by BanzaiMan** [comment]
Can we close this now?

**Comment by oleersoy** [comment]
Yes I'll close it.  I left it open just in case anyone wanted to add to the https://docs.travis-ci.com/user/deployment/pages/ documentation.  I can update it if need be.  Just let me know and thanks again.

**Comment by TAGC** [comment]
I was looking into how I might use travis to auto-deploy to Github Pages. I came across the docs linked above (https://docs.travis-ci.com/user/deployment/pages/) but I wasn't sure if the process would work if I'm git ignoring the generates site content. I had a suspicion I'd have to do something like what @BanzaiMan described but it didn't say whether or not in the docs - I'm glad I was able to find this issue to get confirmation about this.
